+nigeLab/@Tank/subsref.m

```diff
@@ -58,22 +58,54 @@
             '''.'' or ''()'' references.']);
```

Comment:
Something weird happens with `subsref` if I do `addChild` and add a copy of a Block that is on a different parent to the wrong Animal. Then I get errors on the `demo` data using commands you put in the `docs`:
```(matlab)
>> b = tankObj{1,1}; % As expected
>> key = getKey(b,'Public'); % As expected
>> B = tankObj{[1,2],key}; % Sorry I broke it
Index exceeds matrix dimensions.

Error in nigeLab.Animal/subsref>@(x,idx)subsref(x.Children,s(idx,:)) (line 203)
               out = arrayfun(@(x,idx) subsref(x.Children,s(idx,:)),an(:),(1:numel(an))','UniformOutput',false);

Error in nigeLab.Animal/subsref (line 203)
               out = arrayfun(@(x,idx) subsref(x.Children,s(idx,:)),an(:),(1:numel(an))','UniformOutput',false);

Error in nigeLab.Tank/subsref (line 109)
            out = subsref(tankObj.Children,s);
```

Reply:
Even without the `addChild` bit, I am still getting the following error:
```(matlab)
>> key = getKey(b,'Public');
>> B = tankObj{[1,2],key}
Index exceeds matrix dimensions.

Error in nigeLab.Animal/subsref>@(x,idx)subsref(x.Children,s(idx,:)) (line 203)
               out = arrayfun(@(x,idx) subsref(x.Children,s(idx,:)),an(:),(1:numel(an))','UniformOutput',false);

Error in nigeLab.Animal/subsref (line 203)
               out = arrayfun(@(x,idx) subsref(x.Children,s(idx,:)),an(:),(1:numel(an))','UniformOutput',false);

Error in nigeLab.Tank/subsref (line 109)
            out = subsref(tankObj.Children,s);
```

Reply:
Fixed! Nice catch bug catcher.